TPP.Core/Modes/Matchmode.cs

```diff
@@ -137,5 +143,34 @@ public void Dispose()
         {
             _modeBase.Dispose();
         }
+
+        public async void ResetBalances()
```

Comment:
never use `async void` (except for event handlers where not possible otherwise), always use `async Task`, and await it when called